web/client/reducers/map.js

```diff
@@ -113,6 +113,27 @@ function mapConfig(state = null, action) {
     case TOGGLE_UNSAVED_MAP_CHANGES_DIALOG: {
         return assign({}, state, { showUnsavedMapChangesDialog: !(state && state.showUnsavedMapChangesDialog) });
     }
+    case REGISTER_EVENT_LISTENER: {
+        return assign({}, state,
+            {eventListeners: assign({}, state.eventListeners || {},
+                {[action.eventName]: [...(state.eventListeners && state.eventListeners[action.eventName] || []), action.toolName]})});
+    }
+    case UNREGISTER_EVENT_LISTENER: {
+        let data = state;
+        if (state.eventListeners) {
+            const filteredEventNameTools = state.eventListeners[action.eventName].filter(tool => tool !== action.toolName) || [];
+            data = assign({}, state,
+                {eventListeners: assign({}, state.eventListeners,
+                    {[action.eventName]: filteredEventNameTools})});
+        }
+        return data;
+    }
+    case MOUSE_MOVE_MAP_EVENT: {
```

Comment:
this should go in the mouseposition reducer

Reply:
Done